features/agent/ui/AgentChat.tsx

```
@@ -0,0 +1,607 @@
+  userImage?: string;
+  userName?: string;
+}) {
+  const isUser = message.role === "user";
+  const isAssistant = message.role === "assistant";
+  const isStreaming = message.status === "streaming";
+
+  if (!isUser && !isAssistant) return null;
+
+  // Per docs: useSmoothText smooths text as it streams
+  // Pass startStreaming: true when message is actively streaming
+  const [visibleText] = useSmoothText(message.text ?? "", {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)